internal/serialization/serialization.go

```diff
@@ -83,16 +86,20 @@ func (s *Service) ToObject(data *Data) (r interface{}, err error) {
 			err = makeError(rec)
 		}
 	}()
+	var ok bool
 	if data == nil {
 		return nil, nil
 	}
 	typeID := data.Type()
 	if typeID == 0 {
 		return data, nil
 	}
-	serializer, ok := s.registry[typeID]
-	if !ok {
-		return nil, hzerrors.NewHazelcastSerializationError(fmt.Sprintf("there is no suitable de-serializer for type %d", typeID), nil)
+	serializer := s.lookupBuiltinDeserializer(typeID)
```

Comment:
What is the performance impact of this change? We now have to do a bunch of comparisons before reading/writing (especially portable and identified classes), do we have some before and after numbers?

Reply:
There's no corresponding `lookupBuiltinDeserializer` for the `master` branch, so I've added a benchmark for `LookupDefaultSerializer` at: https://github.com/hazelcast/hazelcast-go-client/pull/581/commits/60fb263bd9fec8709430e76a7e26d3cde84db031

You can run it using: `go test -bench BenchmarkService_LookUpDefaultSerializer ./internal/serialization`

According to what I measured, serializer lookup in this PR is faster than the one before:
```
yuce@unreal|13:45:58|521$ go test -bench BenchmarkService_LookUpDefaultSerializer ./internal/serialization
goos: linux
goarch: amd64
pkg: github.com/hazelcast/hazelcast-go-client/internal/serialization
cpu: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
BenchmarkService_LookUpDefaultSerializer-12     25852930                43.15 ns/op
PASS
ok      github.com/hazelcast/hazelcast-go-client/internal/serialization 1.185s
/mnt/d/Work/hazelcast/hazelcast-go-client.git (master)
yuce@unreal|13:46:02|521$ go test -bench BenchmarkService_LookUpDefaultSerializer ./internal/serialization
goos: linux
goarch: amd64
pkg: github.com/hazelcast/hazelcast-go-client/internal/serialization
cpu: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
BenchmarkService_LookUpDefaultSerializer-12     25516767                48.56 ns/op
PASS
ok      github.com/hazelcast/hazelcast-go-client/internal/serialization 1.310s
/mnt/d/Work/hazelcast/hazelcast-go-client.git (master)
yuce@unreal|13:46:07|521$ go test -bench BenchmarkService_LookUpDefaultSerializer ./internal/serialization
goos: linux
goarch: amd64
pkg: github.com/hazelcast/hazelcast-go-client/internal/serialization
cpu: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
BenchmarkService_LookUpDefaultSerializer-12     26558836                43.12 ns/op
PASS
ok      github.com/hazelcast/hazelcast-go-client/internal/serialization 1.215s
/mnt/d/Work/hazelcast/hazelcast-go-client.git (master)
yuce@unreal|13:46:09|521$ go test -bench BenchmarkService_LookUpDefaultSerializer ./internal/serialization
goos: linux
goarch: amd64
pkg: github.com/hazelcast/hazelcast-go-client/internal/serialization
cpu: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
BenchmarkService_LookUpDefaultSerializer-12     24733088                43.92 ns/op
PASS
ok      github.com/hazelcast/hazelcast-go-client/internal/serialization 1.160s
/mnt/d/Work/hazelcast/hazelcast-go-client.git (serialization-improvements)
yuce@unreal|13:46:23|521$ go test -bench BenchmarkService_LookUpDefaultSerializer ./internal/serialization
goos: linux
goarch: amd64
pkg: github.com/hazelcast/hazelcast-go-client/internal/serialization
cpu: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
BenchmarkService_LookUpDefaultSerializer-12     65511451                17.85 ns/op
PASS
ok      github.com/hazelcast/hazelcast-go-client/internal/serialization 2.180s
/mnt/d/Work/hazelcast/hazelcast-go-client.git (serialization-improvements)
yuce@unreal|13:46:27|521$ go test -bench BenchmarkService_LookUpDefaultSerializer ./internal/serialization
goos: linux
goarch: amd64
pkg: github.com/hazelcast/hazelcast-go-client/internal/serialization
cpu: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
BenchmarkService_LookUpDefaultSerializer-12     56741076                18.18 ns/op
PASS
ok      github.com/hazelcast/hazelcast-go-client/internal/serialization 1.079s
```